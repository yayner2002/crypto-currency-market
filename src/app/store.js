import { configureStore } from "@reduxjs/toolkit";
import coinsReducer from "../redux/crypto/Cryptos";

export const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});
