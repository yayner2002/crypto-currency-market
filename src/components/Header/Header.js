import 'bootstrap/dist/css/bootstrap.min.css';
import {} from 'reactstrap';
import 'remixicon/fonts/remixicon.css';
import { Link, useLocation } from 'react-router-dom';
import '../../styles/Header.css';

import React from 'react';

const Header = () => {
  const location = useLocation();
  return (
    <>
      <header className="header">
        <div className="header-left">
          {location.pathname !== '/' ? (
            <span>
              <Link to="/">
                <i className="ri-arrow-left-s-line" />
              </Link>
            </span>
          ) : undefined}
          <span>YayCriptoMarket</span>
        </div>
        <div className="header-center">
          <h5 data-testid="title">Crypto Market</h5>
        </div>
        <div className="header-right">
          <span>
            <i className="ri-mic-line" />
          </span>
          <span>
            <i className="ri-settings-5-fill" />
          </span>
        </div>
      </header>
    </>
  );
};

export default Header;
