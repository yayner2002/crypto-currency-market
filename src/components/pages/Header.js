import 'bootstrap/dist/css/bootstrap.min.css';
import {R} from 'reactstrap';

import React from 'react';

const Header = () => {
  return (
    <div>Header</div>
  )
}

export default Header