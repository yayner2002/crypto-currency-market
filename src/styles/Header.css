.header {
  display: flex;
  justify-content: space-between;
  background: #4369b2;
  color: #fff;
  padding: 1.2rem 0.5rem;
  align-items: center;
  align-content: center;
}

.header-left {
  display: flex;
  justify-content: space-between;
  gap: 0.2rem;
}

.header-left span {
  font-size: 20px;
}

.header-right span {
  cursor: pointer;
  font-size: 18px;
}

.header-center h5 {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-size: 14px;
}

.header-right {
  display: flex;
  gap: 1.2rem;
}
