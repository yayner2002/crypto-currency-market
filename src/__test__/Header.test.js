/* eslint-disable no-unused-vars */
import { toBeInTheDocument } from '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../app/store';
import Header from '../components/Header/Header';

const MockHeader = () => (
  <Provider store={store}>
    <Router>
      <Header />
    </Router>
  </Provider>
);

describe('Test Header component', () => {
  it('should render header component corectlly', () => {
    const tree = renderer.create(<MockHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render crypto title', () => {
    render(<MockHeader />);
    const titleElement = screen.queryByRole('heading', { level: 5 });
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent('Crypto Market');
  });
});
