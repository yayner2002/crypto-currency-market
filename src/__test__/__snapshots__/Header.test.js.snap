// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component should render header component corectlly 1`] = `
<header
  className="header"
>
  <div
    className="header-left"
  >
    <span>
      YayCriptoMarket
    </span>
  </div>
  <div
    className="header-center"
  >
    <h5
      data-testid="title"
    >
      Crypto Market
    </h5>
  </div>
  <div
    className="header-right"
  >
    <span>
      <i
        className="ri-mic-line"
      />
    </span>
    <span>
      <i
        className="ri-settings-5-fill"
      />
    </span>
  </div>
</header>
`;
